src/flatpakRunnerStatusItem.ts

```diff
@@ -12,12 +12,12 @@ export interface Status {
     clickable: Clickable | null
 }
 
-export class StatusBarItem {
+export class FlatpakRunnerStatusItem extends vscode.Disposable {
     private inner: vscode.StatusBarItem
 
-    constructor(extCtx: vscode.ExtensionContext) {
+    constructor() {
+        super(() => this.dispose())
```

Comment:
Not sure I understand the needs for calling dispose on the constructor, is this to make sure the old instance is properly disposed?

Reply:
Its a requirement when extending Disposable. According to vscode API docs, implementing Diposable and pushing it in subscriptions automatically free the resources when the extension deactivates. We could manually push each disposable (like the StatusBarItem) to the subscriptions, but we need to pass ExtensionContext everywhere, which I think is not any better.